### PR TITLE
Preserve whitespace of XML so indent can be calculated correctly

### PR DESCRIPTION
**Pull request by bert2**
# TL;DR

The `XPathDocument` class changes the indentation of the XML when node contents are retrieved via `InnerXml`/`OuterXml`. This leads to problems during indent calculation when `<para>` tags are used inside comments.

# Long explanation

Using `<para>` tags inside XML comments is useful to add line breaks in IntelliSense (which ignores regular line breaks). Here is an example:

image

Now consider the following class where the XML comment of method `Bar()` has `<para>` tags:

```csharp
public class MyClass {
    /// <summary>
    /// First paragraph.
    /// 
    /// Second paragraph.
    /// </summary>
    public void Foo() { }

    /// <summary>
    /// <para>
    /// First paragraph.
    /// </para>
    /// <para>
    /// Second paragraph.
    /// </para>
    /// </summary>
    public void Bar() { }
}
```

This will produce the following XML documentation:

```xml
        <member name="M:DocXmlUnitTests.MyClass.Foo">
            <summary>
            First paragraph.
            
            Second paragraph.
            </summary>
        </member>
        <member name="M:DocXmlUnitTests.MyClass.Bar">
            <summary>
            <para>
            First paragraph.
            </para>
            <para>
            Second paragraph.
            </para>
            </summary>
        </member>
```

When retrieving the comments for `Foo()` the current implementation of DocXml will use the following `OuterXml`...

```xml
<summary>
            First paragraph.
            
            Second paragraph.
            </summary>
```

...to correctly calculate the indentation and fix the `InnerXml` to look like this:

```xml
First paragraph.

Second paragraph.
```

However for `Bar()` the `OuterXml` will look like this:

```xml
<summary>
  <para>
            First paragraph.
            </para>
  <para>
            Second paragraph.
            </para>
</summary>
```

Which leads to an incorrectly calculated indent that in turn causes the result to look like this:

```xml
<para>
            First paragraph.
            </para>
<para>
            Second paragraph.
            </para>
```

With the proposed patch the `OuterXml` will be retrieved exactly as is from the XML file...

```xml
<summary>
            <para>
            First paragraph.
            </para>
            <para>
            Second paragraph.
            </para>
            </summary>
```

...and can then be correctly un-indented to this:

```xml
<para>
First paragraph.
</para>
<para>
Second paragraph.
</para>
```

Note that I'm currently preparing a pull request for mddox which will turn `<para>` tags into line breaks. But for it to work this change is needed in DocXml.

I know... lots of text for such a small PR, but I felt a proper explanation was necessary.

### CODE REVIEW

**Comment by loxsmoke** [comment]
Good explanation. Good fix. I'll hold the new version release for a couple of days in case something else comes up.
